### PR TITLE
Rethink xLayers's UX

### LINKED ISSUE #379: Rethink xLayers's UX
**Is your feature request related to a problem? Please describe.**
- Currently, when users open the landing page at xlayers.dev, they can read about what's xlayers. 
- Next, they click on get started, and get redirected to the [upload](https://xlayers.dev/upload) page.
- After uploading a design file, they get redirected to [the preview UI](https://xlayers.dev/#/editor/preview) where they can inspect their design.
- If they want to generate code from the design, [from the code generation UI](https://xlayers.dev/#/editor/code), xlayers generates code for ALL supported framework at once!

This could be problematic when we will have a lot of supported frameworks. We need to simply this workflow.

**Describe the solution you'd like**
- On the landing page, we should allow users to choose their framework they wanna generate code for.
- When clicking on the framework, they will get redirected to the upload UI to upload their design.
- The rest of the workflow should be the same, expect...
- The code generation UI now only show the selected framework.
  - Note: we should allow users to instantly switch the selected framework (from the code generation UI).

### PR DESCRIPTION
**Pull request by ayshiff**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: Closes #379 

This PR updates the current UX workflow.

This is still a work in progress because I need to do some **adjustments, refactoring and add some tests**.
I want to have your point of view on my implementation.

## What is the new behavior?

On the landing page, users can choose their framework they wanna generate code for.
On the code generation UI, we only show the selected framework with the possibility to switch to other frameworks.

### Landing Page 

<p align="center">
Landing Page
</p>

### Code generation

<p align="center">
Code generation 1
</p>

<p align="center">
Code generation 2
</p>

### [EDIT] Header style updated in [last commit](https://github.com/xlayers/xlayers/pull/395/commits/03aff0eb7ebced3b6b27e46126c08b5efa1f7bfe)

<p align="center">
Code generation 1
</p>

## New workflow

<p align="center">
New workflow
</p>

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by Jefiozie** [comment]
Hi @anediaz would you review this PR?

**Review by anediaz — COMMENTED** [review]
LGTM

**Comment by anediaz** [comment]
The code review is OK for me, @manekinekko & @Jefiozie, could you please take a look and give your approval so it can be merged?
I can't wait to see the new UX present in xlayers.dev :-)

File: apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts [hunk 1]
```diff
@@ -19,65 +18,37 @@ const githubIssueLink =
...
-
-  constructor(private codegen: CodeGenService, private readonly store: Store) {}
-
-  ngOnInit() {}
+  public frameworks: UICodeGen[] = codeGenList;
 
-  ngAfterContentInit() {
-    this.generateAngular();
-  }
+  public selectedFramework: UICodeGen = codeGenList[0];
```
Comment (Jefiozie) [inline apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts:27]:
```suggestion
  public selectedFramework: UICodeGen = frameworks[0];
```

File: apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts [hunk 2]
```diff
@@ -19,65 +18,37 @@ const githubIssueLink =
...
-  generateVue() {
-    this.codeSetting = this.codegen.generate(CodeGenKind.Vue);
-    this.updateState();
-  }
-
-  generateWc() {
-    this.codeSetting = this.codegen.generate(CodeGenKind.WC);
-    this.updateState();
-  }
+  constructor(private codegen: CodeGenService, private readonly store: Store) {}
```
Comment (Jefiozie) [inline apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts:29]:
```suggestion
  constructor(private readonly codegen: CodeGenService, private readonly store: Store) {}
```

File: apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts [hunk 3]
```diff
@@ -19,65 +18,37 @@ const githubIssueLink =
...
 
-  generateStencil() {
-    this.codeSetting = this.codegen.generate(CodeGenKind.Stencil);
-    this.updateState();
+  ngOnInit() {
+    this.store.select(CodeGenState.codegen).subscribe((codegen) => {
+      if (codegen.kind) {
```
Comment (Jefiozie) [inline apps/xlayers/src/app/editor/code/editor-container/editor-container.component.ts:33]: Could `kind` be null/ falsy? If that is the case wouldn't line 38  file also?

File: apps/xlayers/src/home/landing/landing.component.ts [hunk 4]
```diff
@@ -159,7 +163,15 @@ export class LandingComponent implements OnInit {
...
 
-  constructor() {}
+  public frameworks: UICodeGen[] = codeGenList;
+
+  constructor(private router: Router, private readonly store: Store) {}
 
   ngOnInit() {}
+
+  selectFramework(framework: CodeGenKind) {
+    console.log(framework);
```
Comment (Jefiozie) [inline apps/xlayers/src/home/landing/landing.component.ts:173]: I think this line can be removed

**Review by Jefiozie — CHANGES_REQUESTED** [review]
I think we are almost there, some small comments that we need to handle before merging this one in. I love the improvements you've made to the UI! Thank you 🙏🏻

**Comment by ayshiff** [comment]
Thank you for your comments !

I have a comment to make about the `.svg` framework icons.
The different sizes of framework icons on the landing page confuses me a bit but I don't know if we should correct this in this PR.
The icons should be resized so that they have the same height. 
I'll let you tell me if it's something that needs to be fixed or not.

**Comment by anediaz** [comment]
Good point! I agree with you!

I've looked to the different .svg files we have and effectively, some of them have a defined ``height``and ``width``.
It'd be quite simple and fast to remove them, in order they have all the same initial size.

I think it is worth fixing it :-)

**Comment by ayshiff** [comment]
> Good point! I agree with you!
> 
> I've looked to the different .svg files we have and effectively, some of them have a defined `height`and `width`.
> It'd be quite simple and fast to remove them, in order they have all the same initial size.
> 
> I think it is worth fixing it :-)

The problem was a little more complicated than that.
Removing the `height` and `width` didn't change the problem as the issue comes from the dimensions of what's inside the svg (`<g>` or `<path>`)
I resized the icons using a tool and now they have similar dimensions 👍

**Comment by ayshiff** [comment]
@anediaz @Jefiozie Does everything look good to you?

**Review by Jefiozie — APPROVED** [review]
LGTM

**Comment by manekinekko** [comment]
Thank you so much @ayshiff for your contribution ❤️
After iterating over your PR, I made a few changes based on your design to the landing page.

@Jefiozie @anediaz let me know what you think before I merge these changes to main.

image

**Review by manekinekko — APPROVED** [review]

**Comment by Jefiozie** [comment]
Okay, I reallly like this 🤩

LGTM!!! 🎉

**Comment by anediaz** [comment]
I love the design, I find it very original :tada: :star_struck: 
Good work !!!

**Comment by manekinekko** [comment]
Awesome job @ayshiff. Your code was merged in c1adf5418715929d547c2132d0aeb3b2120c68bd. Congratulations!

**Pull request closed without merging**